src/bin/tikv-ctl.rs

```diff
@@ -182,3 +313,126 @@ fn dump_range(db: DB, from: String, to: Option<String>, limit: Option<u64>, cf:
+            })
+            .collect();
+        let default_value: Vec<_> = test_data.iter().map(|data| *data.1).collect();
+        let kvs = keys.iter().zip(default_value.iter());
+        for (k, v) in kvs {
+            db.put(k.as_slice(), v).unwrap();
+        }
+        let kvs_gen: Vec<MvccKv<Vec<u8>>> = gen_mvcc_iter(&db, "k", CF_DEFAULT).unwrap();
+        let mut test_iter = test_data.clone();
+        assert_eq!(test_iter.len(), test_data.len());
+        for kv in kvs_gen {
+            let ts = kv.key.decode_ts().unwrap();
+            let key = kv.key.truncate_ts().unwrap().raw().unwrap();
+            let value = kv.value.as_slice();
+            test_iter.retain(|&s| !(&s.0[..] == key.as_slice() && &s.1[..] == value && s.2 == ts));
+        }
+        assert_eq!(test_iter.len(), 0);
+
+        // Test MVCC Lock
+        let test_data_lock = vec![(b"kv", LockType::Put, b"v", 5),
+                                  (b"kx", LockType::Lock, b"x", 10),
+                                  (b"kz", LockType::Delete, b"y", 15)];
+        let keys: Vec<_> = test_data_lock.iter()
+            .map(|data| {
+                let encoded = encode_bytes(&data.0[..]);
+                keys::data_key(encoded.as_slice())
+            })
+            .collect();
+        let lock_value: Vec<_> = test_data_lock.iter()
+            .map(|data| Lock::new(data.1, data.2.to_vec(), data.3).to_bytes())
+            .collect();
+        let kvs = keys.iter().zip(lock_value.iter());
+        let lock_cf = db.cf_handle(CF_LOCK).unwrap();
+        for (k, v) in kvs {
+            db.put_cf(lock_cf, k.as_slice(), v.as_slice()).unwrap();
+        }
+        let kvs_gen: Vec<MvccKv<Lock>> = gen_mvcc_iter(&db, "kv", CF_LOCK).unwrap();
+        assert_eq!(kvs_gen.len(), 1);
+        let kv = &kvs_gen[0];
+        let lock = &kv.value;
+        let key = kv.key.raw().unwrap();
+        let test_data = &test_data_lock[0];
+        assert!(&key[..] == test_data.0 && test_data.1 == lock.lock_type &&
+                &test_data.2[..] == &lock.primary[..] && test_data.3 == lock.ts);
+        let kvs_gen: Vec<MvccKv<Lock>> = gen_mvcc_iter(&db, "kx", CF_LOCK).unwrap();
+        assert_eq!(kvs_gen.len(), 1);
+        let kv = &kvs_gen[0];
+        let lock = &kv.value;
+        let key = kv.key.raw().unwrap();
+        let test_data = &test_data_lock[1];
+        assert!(&key[..] == test_data.0 && test_data.1 == lock.lock_type &&
+                &test_data.2[..] == &lock.primary[..] && test_data.3 == lock.ts);
+        let kvs_gen: Vec<MvccKv<Lock>> = gen_mvcc_iter(&db, "kz", CF_LOCK).unwrap();
+        assert_eq!(kvs_gen.len(), 1);
+        let kv = &kvs_gen[0];
+        let lock = &kv.value;
+        let key = kv.key.raw().unwrap();
+        let test_data = &test_data_lock[2];
+        assert!(&key[..] == test_data.0 && test_data.1 == lock.lock_type &&
+                &test_data.2[..] == &lock.primary[..] && test_data.3 == lock.ts);
```

Comment:
Too many duplicated lines for the lock check. I think we may extract a assert function.